### DISCUSSION TITLE
Logarithm functions with different bases

### DISCUSSION THREAD

**Discussion by jrom99** [lcn2/calc#86] [Q&A] [answered]
Hello, I'm trying to use calc to calculate a logarithm in base 2, but I'm having trouble finding a function that accepts a logarithm with a different base (similar to `math.log` in python). Is such a function available in calc? Where can I import it from? I'm aware that I could use the formula `logb(x) = log(x)/log(b)`, but a general function would be easier.

Thanks in advance!

**Comment by lcn2** [comment]
> Hello, I'm trying to use calc to calculate a logarithm in base 2, but I'm having trouble finding a function that accepts a logarithm with a different base (similar to `math.log` in python). Is such a function available in calc? Where can I import it from? I'm aware that I could use the formula `logb(x) = log(x)/log(b)`, but a general function would be easier.
> 
> 
> 
> Thanks in advance!

We could certainly consider adding both a log base 2 as well a a more general log base n builtin functions for version 2.14.4.

Perhaps by adding a `lb()` function for log base 2 and some other builtin there the 2nd arg would be a logarithm base.

**Reply by jrom99** [reply]
In this case, may I suggest using `log2` for the log base 2 and `logn` for the arbitrary base?

**Reply by lcn2** [reply]
> In this case, may I suggest using `log2` for the log base 2 and `logn` for the arbitrary base?

Sure, `log2()` and `logn()` are fine choices.

**Comment by lcn2** [comment]
We have added a priority enhancement issue #87, @jrom99.  Feel free to comment on that issue.

**Comment by lcn2** [comment]
Given the new enhancement issue https://github.com/lcn2/calc/issues/87, @jrom99, we will now close this discussion with our thanks.   Feel free to comment on that issue if you wish.

**Comment by lcn2** [comment, marked as answer]
FYI: With commit 1c839dfedec07f70d5ce07a21028c3517ec94a30, the log2(x) and logn(x, n) builtin functions have been added, @jrom99 

We expect to release calc version 2.15.x with this code in the near future.  See issue #87 for details.
